lab-trent/routes/user-route.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const express = require('express');
+const jsonToken = require('jsonwebtoken');
+const createError = require('http-errors');
+const constants = require('../lib/constants');
+
+const User = require('../models/user');
+
+const authMiddleware = require('../lib/authentication');
+
+const router = module.exports = express.Router();
+
+router.post('/users', function(req, res, next) {
+  new User(req.body).save().then(function(user) {

```

Comment:
So you've got a pre-save hook that's taking care of the pw hashing? I'm reading that correct?

Nifty!